### DISCUSSION TITLE
When using rabbitmq.conf to configure ldap, rabbitmq-server cannot start because the ldap organization exists in Chinese.

### DISCUSSION THREAD

**Discussion by lcy0828** [rabbitmq/rabbitmq-server#10960] [Questions] [answered]
### Describe the bug

When using rabbitmq.conf to configure ldap, rabbitmq-server cannot start because the ldap organization exists in Chinese.

### Reproduction steps

1.修改rabbitmq.conf
2.添加中文
3.19:50:36.317 [error] /etc/rabbitmq/rabbitmq.conf: Error converting value on line #6 to latin1

### Expected behavior

auth_ldap.user_dn_pattern= cn=${username},ou=部门, ou=分公司 , ou=总公司 , dc=corp , dc=xxxx , dc=com

**Comment by lukebakken** [comment]
Hello, thanks for using RabbitMQ.

You _MUST_ provide more information before we can consider this a bug to address. Please provide the following:
* Your RabbitMQ and Erlang version
* Which operating system and version you use to run RabbitMQ
* Please **ATTACH** your **FULL, UN-EDITED** `rabbitmq.conf` file to this discussion or to a comment. If you don't provide the complete file, or if you paste the contents to a comment (rather than attaching the file), we can't assist you. This is important because we need to be certain we have the exact same file you are using.

**Reply by lcy0828** [reply]
rabbitmq-server-3.7.8-1.el7.noarch.rpm 
erlang-21.3.8.6-1.el7.x86_64.rpm
CentOS Linux release 7.9.2009 (Core)

rabbitmq.conf:
auth_backends.1 = internal
auth_backends.2 = ldap
auth_ldap.servers.1 =  ldaprw.xxx.xxx.com
auth_ldap.port = 389
auth_ldap.use_ssl = false
auth_ldap.user_dn_pattern= cn=${username},ou=事业部 , ou=新业务 , ou=总公司 , dc=xxx, dc=xxx, dc=com

**Comment by michaelklishin** [comment]
This could be a limitation in the `rabbitmq.conf` parser, [Cuttlefish](https://github.com/Kyorai/cuttlefish). It does not seem to be something specific to LDAP, although we have no executable way to reproduce.

Setting a configuration key such as `cluster_name` to a similar value would be a good way to prove this hypothesis.

You are welcome to investigate this behavior and contribute an improvement w.r.t. CJK/non-ASCII character handling, or sponsor its development. Open source does not mean "free labor you can ask to solve your problem on GitHub."

**Reply by lcy0828** [reply]
rabbitmq-server-3.7.8-1.el7.noarch.rpm 
erlang-21.3.8.6-1.el7.x86_64.rpm
CentOS Linux release 7.9.2009 (Core)

rabbitmq.conf:
auth_backends.1 = internal
auth_backends.2 = ldap
auth_ldap.servers.1 =  ldaprw.xxx.xxx.com
auth_ldap.port = 389
auth_ldap.use_ssl = false
auth_ldap.user_dn_pattern= cn=${username},ou=事业部 , ou=新业务 , ou=总公司 , dc=xxx, dc=xxx, dc=com

**Comment by mkuratczyk** [comment]
RabbitMQ 3.7 has been out of support for many years. You should be using 3.12 or 3.13 at this point.
https://www.rabbitmq.com/release-information

I quickly checked that I can start 3.13 with this config file, so most likely this issue was fixed a long time ago.

**Reply by lcy0828** [reply]
In fact, what I want to do is to distinguish the user's permissions. Now I have a scenario where I need to distinguish the user's creation and deletion permissions. Only give the user permissions to read, write, and create, and take back the permission to delete. In the official documentation In fact, it is mentioned that fine-grained permissions include creation and deletion, but in fact I cannot make a distinction between creation and deletion. I would like to ask if this requires any special configuration.
![image](https://github.com/rabbitmq/rabbitmq-server/assets/48135114/1929c4e8-2e8b-4948-ad65-6d0035065b1d)

**Reply by mkuratczyk** [reply, marked as answer]
I don't see how this is related to the original question. The `configure` permissions controls both creation and deletion. I don't see how you read from this table that these are separate permissions.

**Comment by michaelklishin** [comment]
Closing and locking because

1. We do not provide general LDAP-related support to non-paying users
2. RabbitMQ 3.7 has reached EOL years ago
